org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/xsd/XSDValidationExtensionsTest.java

```diff
@@ -348,6 +348,39 @@ public void src_import_1_2_different_range() throws BadLocationException {
 		testCodeActionsFor(xml, d, ca(d, te(2, 11, 2, 11, " namespace=\"\"")), ca(d, te(1, 54, 1, 54, " targetNamespace=\"\"")));
 	}
 
+	@Test
+	public void schema_reference_4_BadSchemaLocation() throws BadLocationException {
+		String xsd = "<xs:schema xmlns:xs='http://www.w3.org/2001/XMLSchema'>\r\n" +
+		"<xs:import namespace='http://foo' schemaLocation='bad.xsd' />\r\n" +
+		"<xs:element name='foo'></xs:element>\r\n" +
+		"</xs:schema>";
+
+		Diagnostic d = d(1, 49, 1, 58, XSDErrorCode.schema_reference_4);
+		testDiagnosticsFor(xsd, d);
+	}
+
+	@Test
+	public void schema_reference_4_GoodSchemaLocation() throws BadLocationException {

```

Comment:
The correct test is:

```java
	@Test
	public void schema_reference_4_GoodSchemaLocation() throws BadLocationException {
		String xsd = "<xs:schema xmlns:xs='http://www.w3.org/2001/XMLSchema'>\r\n" +
		"<xs:import namespace='team_namespace' schemaLocation='src/test/resources/xsd/team.xsd' />\r\n" +
		"<xs:element name='foo'></xs:element>\r\n" +
		"</xs:schema>";
		testDiagnosticsFor(xsd);
	}
```

It should have none error. Here team_namespace is used because team.xsd define it as targetNamespace.

In otherwords it should be nice to write a test with bad namespace 

```java
@Test
	public void schema_reference_4_GoodSchemaLocation() throws BadLocationException {
		String xsd = "<xs:schema xmlns:xs='http://www.w3.org/2001/XMLSchema'>\r\n" +
		"<xs:import namespace='BAD_NAMESPACE' schemaLocation='src/test/resources/xsd/team.xsd' />\r\n" +
		"<xs:element name='foo'></xs:element>\r\n" +
		"</xs:schema>";
	... should have some error
	}
```

Reply:
From what I'm gathering, I believe the bad namespace error is for `src-import.3.1`, which isn't listed in any error file. Should I add this?

Reply:
Tes please define it manage the error range and write a test